tests/utils/test_requirements_utils.py

```diff
@@ -257,3 +260,13 @@ def test_get_pinned_requirement_local_version_label(tmpdir):
             f"Found my_package version (1.2.3+{lvl}) contains a local version label (+{lvl})."
         )
     assert req == "my_package==1.2.3"
+
+
+def test_infer_requirements_excludes_mlflow():
+    with mock.patch(
+        "mlflow.utils.requirements_utils._capture_imported_modules",
+        return_value=["mlflow", "pytest"],
+    ):
+        mlflow_package = "mlflow-skinny" if "MLFLOW_SKINNY" in os.environ else "mlflow"
+        assert mlflow_package in _module_to_packages("mlflow")
+        assert _infer_requirements("/path/to/model", "mlflow") == [f"pytest=={pytest.__version__}"]
```

Comment:
@dbczumar Added a test to ensure `_infer_requirements` excludes `mlflow-skinny` and `mlflow`.

Reply:
This test executed here: https://github.com/mlflow/mlflow/pull/4662/files#diff-3a3541631d60e418ecd3374eae7350595946e0b7c8694ef15a9c6549f4a3af99R35